### DISCUSSION TITLE
是否支持只读属性和自定义构造函数？

### DISCUSSION THREAD

**Discussion by CodingOctocat** [dotnetcore/FreeSql#1823] [Q&A]
我最近在使用 sqlite/orm 时遇到一些问题：

1. sqlite.net
    - 它必须要有默认构造函数
    - 不支持继承而来的属性
    - 不支持只读属性
2. linq2db
    - 如果存在多个构造函数但没有默认构造函数，将引发异常，即：只支持只有一个自定义构造函数(可private)的情况和多构造函数但必须有空构造函数
    - 支持继承而来的属性，但可写，不可读，即：写入到数据库的继承属性无法反序列化为对象，即使我构造函数中有赋值
    - 支持只读属性，但仅限只有一个构造函数且构造函数有为只读属性赋值的情况

**Comment by 2881099** [comment]
没有特意去测试，不知道是否支持，建议实体类型简洁为主

**Comment by 2881099** [comment]
属性可以设置 protected set 达到只读的目的。
